### PR TITLE
Cleanup all implicit bool conversions

### PR DESCRIPTION
**Pull request by AJenbo**
With this `readability-implicit-bool-conversion` will be fully applied :)

### CODE REVIEW

File: Source/monster.cpp [hunk 1]
```diff
@@ -2603,20 +2603,17 @@ bool M_DoDeath(int i)
...
-		if (x < 0)
-			x = -1;
-		else
-			x = x > 0;
-		ViewX += x;
+		if (monster[i].position.tile.x > ViewX) {
+			ViewX--;
+		} else if (monster[i].position.tile.x < ViewX) {
+			ViewX++;
+		}
```
Comment (StephenCWills) [inline Source/monster.cpp:2610]: Using substitution:
```
x < 0
monster[i].position.tile.x - ViewX < 0
monster[i].position.tile.x < ViewX
```

```suggestion
		if (monster[i].position.tile.x < ViewX) {
			ViewX--;
		} else if (monster[i].position.tile.x > ViewX) {
			ViewX++;
		}
```

File: Source/monster.cpp [hunk 2]
```diff
@@ -2603,20 +2603,17 @@ bool M_DoDeath(int i)
...
 
-		y = monster[i].position.tile.y - ViewY;
-		if (y < 0) {
-			y = -1;
-		} else {
-			y = y > 0;
+		if (monster[i].position.tile.y > ViewY) {
+			ViewY--;
+		} else if (monster[i].position.tile.y < ViewY) {
+			ViewY++;
```
Comment (StephenCWills) [inline Source/monster.cpp:2615]:
```suggestion
		if (monster[i].position.tile.y < ViewY) {
			ViewY--;
		} else if (monster[i].position.tile.y > ViewY) {
			ViewY++;
```

File: Source/monster.cpp [hunk 3]
```diff
@@ -2846,29 +2843,34 @@ bool M_DumbWalk(int i, Direction md)
...
 
-	Direction mdtemp = md;
 	bool ok = DirOK(i, md);
-	if (!ok) {
-		if (*dir) {
-			md = right[mdtemp];
-			ok = DirOK(i, md) || (md = right[right[mdtemp]], DirOK(i, md));
+
+	for (int j = 0; !ok && j < 2; j++) {
+		// Keep trying up to an including backwards
```
Comment (StephenCWills) [inline Source/monster.cpp:2856]: This looks to me like it's turning the opposite direction from the original 90-degree turn. So it's not turning the monster backwards, but rather forwards.

```suggestion
		// Keep trying by turning back toward the original direction
```

File: Source/monster.cpp [hunk 4]
```diff
@@ -2846,29 +2843,34 @@ bool M_DumbWalk(int i, Direction md)
...
+		ok = DirOK(i, md);
 	}
-	if (ok) {
-		M_WalkDir(i, md);
-	} else {
-		*dir = !*dir;
-		ok = M_CallWalk(i, opposite[mdtemp]);
+
+	if (!ok) {
+		// Try 90 degrees in the oposit then desired direction
```
Comment (StephenCWills) [inline Source/monster.cpp:2866]: Just grammar and spelling here.

```suggestion
		// Try 90 degrees in the opposite than desired direction
```

**Review by StephenCWills — CHANGES_REQUESTED** [review]

File: Source/monster.cpp (hunk 3 above)
Comment (AJenbo) [reply Source/monster.cpp:2856]: Oh you are right, I might rethink how this is done in that case.

**Review by StephenCWills — DISMISSED** [review]
It looks correct now, but I wonder if you might like it better if we reduce nesting, like below. This feels a bit more straightforward to me.

```C++
bool M_RoundWalk(int i, Direction direction, int *dir)
{
	Direction turn45deg = turn(direction, *dir != 0);
	Direction turn90deg = turn(turn45deg, *dir != 0);

	if (DirOK(i, turn90deg)) {
		// Turn 90 degrees
		M_WalkDir(i, turn90deg);
		return true;
	}

	if (DirOK(i, turn45deg)) {
		// Only do a small turn
		M_WalkDir(i, turn45deg);
		return true;
	}

	if (DirOK(i, direction)) {
		// Continue straight
		M_WalkDir(i, direction);
		return true;
	}

	// Try 90 degrees in the opposite than desired direction
	*dir = (*dir == 0) ? 1 : 0;
	return M_CallWalk(i, opposite[turn90deg]);
}
```

**Comment by AJenbo** [comment]
> I wonder if you might like it better if we reduce nesting

Yeah I had an inner conflict where I couldn't decide if I would avoid the repetition or nesting. You suggestion of utilizing the direction variables directly does improve both so I think that is indeed the best solution here.

**Pull request merged**